src/Microsoft.ML.Data/Transforms/ColumnBindingsBase.cs

```diff
@@ -14,16 +14,40 @@
 
 namespace Microsoft.ML.Data
 {
-    internal abstract class SourceNameColumnBase
+    public class OneToOneColumn
     {
+        /// <summary>Name of the column resulting from the transformation of <see cref="InputColumnName"/>.</summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Name of the new column", ShortName = "name")]
-        public string Name;
+        public string OutputColumnName;
+
+        /// <summary>Name of the column to transform. If set to <see langword="null"/>, the value of the <see cref= "OutputColumnName"/> will be used as source.</summary>
+        [Argument(ArgumentType.AtMostOnce, HelpText = "Name of the source column", ShortName = "source, src")]
+        public string InputColumnName = null;
 
-        [Argument(ArgumentType.AtMostOnce, HelpText = "Name of the source column", ShortName = "src")]
-        public string Source;
+        /// <summary>
+        /// Instantiates a <see cref="ColumnOptions"/> from a tuple of input and output column names.
+        /// </summary>
+        public static implicit operator OneToOneColumn((string outputColumnName, string inputColumnName) value)
+        {
+            return new OneToOneColumn(value.outputColumnName, value.inputColumnName);
+        }
+
+        /// <summary>
+        /// Instantiates a <see cref="ColumnOptions"/> from the column name.
+        /// </summary>
+        public static implicit operator OneToOneColumn(string outputColumnName)
+        {
+            return new OneToOneColumn(outputColumnName);
+        }
+
+        public OneToOneColumn(string outputColumnName, string inputColumnName = null)
```

Comment:
>OneToOneColumn [](start = 15, length = 14)

Note that we have a non-sealed directly instantiable class...

Reply:
Ok will remove the constructor

---
In reply to: [267546107](https://github.com/dotnet/machinelearning/pull/3033#discussion_r267546107) [](ancestors = 267546107)